### DISCUSSION TITLE
Any ideas let toga android run in backgroud?

### DISCUSSION THREAD

**Discussion by ThinkNaive** [beeware/toga#1679] [Q&A] [answered]
I wrote a tiny app downloading files. Every time the screen off, the app stops downloading (may be hang by system strategy). Are there any approaches (e.g. wake lock or play silent sound) can let it run in background? Thanks.

**Comment by mhsmith** [comment, marked as answer]
You shold be able to use a wake lock, [like this](https://developer.android.com/training/scheduling/wakelock#cpu). Notes on converting the code to Python:

* If `self` is your `App` object, then the `Context` on which to call `getSystemService` is `self._impl.native`. 
* `POWER_SERVICE` can also be obtained as an attribute of the `Context`.
* `PowerManager` can be imported using `from ... import ...` syntax, as shown [here](https://chaquo.com/chaquopy/doc/current/python.html#import-hook).
